go/mcap/write_sizer.go

```diff
@@ -15,9 +14,9 @@ func (w *writeSizer) Write(p []byte) (int, error) {
 	return w.w.Write(p)
 }
 
-func newWriteSizer(w io.Writer) *writeSizer {
+func newWriteSizer(w io.Writer, calculateCRC bool) *writeSizer {
 	return &writeSizer{
-		w: newCRCWriter(w),
+		w: newCRCWriter(w, calculateCRC),
```

Comment:
It seems a bit weird to pass a boolean to calculateCRC to crc-writer. How about this?

```
if (calculateCRC) {
  w = newCRCWriter(w)
}
return &writeSizer{w: w}
```